### ISSUE TITLE
unexpected GraphQL error encountered in Dagit Configuration tab

### ISSUE THREAD

**Issue by lynma01** [dagster-io/dagster#3955]
## Summary
Created pipeline's first mode_defs by setting name without setting any other args. While the repo is recognized by Dagit, and loads without issue, when I try to reach the Configuration tab, I get the error: "unexpected GraphQL error"

I'm unsure whether this has to do with the mode_defs directly. I vaguely remember encountering this issue once before but not the steps I took to produce it that first time.

**Expected Behvaior**: set pipeline mode_defs and then configure logger_defs from Dagit Configuration

**Actual Behavior**: "unexpected GraphQL error"

## Reproduction
```python
@pipeline(
    mode_defs=[
        ModeDefinition(name="local"),
        ModeDefinition(name="prod")
    ]
)
```

-> Open Dagit -> Status -> Configuration Tab -> unexpected GraphQL error

## Error Log
 
Operation name: InstanceConfigQuery

Message: dump_all() got an unexpected keyword argument 'sort_keys'

Path: ["instance","info"]

Locations: [{"line":4,"column":5}]

Stack Trace:
  File "/home/matt/.local/lib/python3.9/site-packages/graphql/execution/executor.py", line 452, in resolve_or_error
    return executor.execute(resolve_fn, source, info, **args)
  File "/home/matt/.local/lib/python3.9/site-packages/graphql/execution/executors/sync.py", line 16, in execute
    return fn(*args, **kwargs)
  File "/home/matt/.local/lib/python3.9/site-packages/dagster_graphql/schema/instance.py", line 102, in resolve_info
    return self._instance.info_str()
  File "/home/matt/.local/lib/python3.9/site-packages/dagster/core/instance/__init__.py", line 485, in info_str
    return yaml.dump(self.info_dict(), default_flow_style=False, sort_keys=False)
  File "/usr/lib/python3/dist-packages/yaml/__init__.py", line 200, in dump
    return dump_all([data], stream, Dumper=Dumper, **kwds)

---

### Error Screen
![image](https://user-images.githubusercontent.com/11153510/112675510-9fcb9280-8e3d-11eb-978a-3abe24aef673.png)

### Loaded Repo
![image](https://user-images.githubusercontent.com/11153510/112675602-c093e800-8e3d-11eb-938b-2ac7ef5e3eeb.png)

## Additional Info about Your Environment
Linux matt-VirtualBox 5.4.0-70-generic #78~18.04.1-Ubuntu SMP Sat Mar 20 14:10:07 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

**Comment by alangenfeld** [comment]
odd - can you run `pip list | grep -i yaml`? I wonder if some old (or new?) version of pyyaml doesnt have `sort_keys` in `dumps`

**Comment by lynma01** [comment]
@alangenfeld I fixed the issue from your conjecture about pyyaml- you were exactly right.

Long story short, I needed to re-pip install requirements.txt after a series of dependa-bot alerts updating me from pyYAML 3.12 -> 5.4 among other things. I had set Github Actions to auto-merge those medium/high security PR's, but I neglected to actually update my python venv.

PyYAML 3.12 *I think* is the version that was giving me the GraphQL Error since that was the one I was unwittingly operating on this whole time.

![image](https://user-images.githubusercontent.com/11153510/112754280-5b5d0580-8fa9-11eb-86a5-42faa1baf532.png)

And then, after updating from 3.12 -> 5.4

![image](https://user-images.githubusercontent.com/11153510/112754567-9dd31200-8faa-11eb-8e57-c0c088acbeb0.png)
